sml_small/editing/thousand_pounds/thousand_pounds.py

```diff
@@ -76,24 +121,89 @@ def run(
     ) as error:  # Catch any underlying errors and return a coherent output dataset
         # Ensure we populate the output target variables with the same output values as originally given
         target_variables_final = []
-        for question in target_variables:
-            target_variables_final.append(
-                Target_variable(
-                    identifier=question.identifier,
-                    original_value=question.original_value,
-                    final_value=question.original_value,
-                )
-            )
+        if principal_identifier is None:
+            principal_identifier = "N/A"
+        raise TPException(f"identifier: {principal_identifier}", error)
 
-        return Thousands_output(
-            principal_identifier=principal_identifier,
-            principal_original_value=principal_variable,
-            principal_final_value=principal_variable,  # Always return the final output as the same as the input
-            target_variables=target_variables_final,
-            tpc_ratio=None,
-            tpc_marker="E",
-            error_description=f"{error}",
-        )
+
+def validate_input(
+    predictive: Optional[float],
+    auxiliary: Optional[float],
+    principal_variable: float,
+    lower_limit: float,
+    upper_limit: float,
+    target_variables: List[Target_variable],
+) -> Tuple[
+    Union[float, None],
+    Union[float, None],
+    float,
+    float,
+    float,
+    List[Target_variable],
+]:
+    """
+    This function is used to validate the data passed to the thousand_pounds
+    method ensuring that the values are present when expected and that they are of
+    the correct type. If invalid data is received then an appropriate exception is
+    raised.
+
+    :param predictive:  Value used for 'previous' response (Returned/Imputed/Constructed)
+    :type predictive: Optional[float]
+    :param auxiliary: Calculated response for the 'previous' period
+    :type auxiliary: Optional[float]
+    :param principal_variable: Original response value provided for the 'current' period
+    :type principal_variable: float
+    :param lower_limit: Lower bound of 'error ratio' threshold
+    :type lower_limit: float
+    :param upper_limit: Upper bound of 'error ratio' threshold
+    :type upper_limit: float
+    :param target_variables: List of monetary variables that may be automatically corrected
+    :type target_variables: List[Targer_variable]
+
+    :return: predictive is returned as a converted float
```

Comment:
I don't think we are converting to float on the return so the comments here are not quite right.